apps/site/assets/ts/schedule/components/line-diagram/LineDiagram.tsx

```diff
@@ -168,104 +170,143 @@ const LineDiagram = ({
     branchedLineDiagram = lineDiagram;
   }
 
+  const [stopQuery, setStopQuery, filteredStops] = useFilteredList(
+    lineDiagram,
+    "route_stop.name"
+  );
+
   return (
     <>
       <h3 className="m-schedule-diagram__heading">
         {routeType === 0 || routeType === 1 || routeType === 2
           ? "Stations"
           : "Stops"}
       </h3>
-      <div
-        className={`m-schedule-diagram m-schedule-diagram--${treeDirection}`}
-      >
-        {branchedLineDiagram.map(
-          (stopOrStops: LineDiagramStop | LineDiagramStop[], bldIndex) => {
-            if (Array.isArray(stopOrStops)) {
-              if (stopOrStops.length > 2) {
-                let willMerge = false;
-                const adjacentIndex =
-                  treeDirection === "inward" ? bldIndex + 1 : bldIndex - 1;
-                const adjacentStop = branchedLineDiagram[
-                  adjacentIndex
-                ] as LineDiagramStop;
-                if (adjacentStop && !Array.isArray(adjacentStop)) {
-                  willMerge = mergeIndices.includes(
-                    lineDiagram.indexOf(adjacentStop)
+      <SearchBox
+        id="stop-search"
+        labelText="Search for a stop"
```

Comment:
This text always says "Search for a stop", but the header sometimes calls them "Stations".